### PR TITLE
support image pull secrets from fluid helm charts to alluxio runtime …

### LINKED ISSUE #2174: [FEATURES] support  alluxio  image pull secrets
**What feature you'd like to add:**
in this pr [support image pull secrets](https://github.com/fluid-cloudnative/fluid/pull/2038/) , only support image pull secrets when pulling pods in fluid-system , but l want to pull own alluxio image that need to provide image pull secrets.

**Why is this feature needed:**
In our scenario, we need to use our own alluxio. so I think it is necessary to support the image  pull the secrets when alluxio runtime is created

### PR DESCRIPTION
**Pull request by allenhaozi**
…controller

Signed-off-by: mahao <allenhaozi@gmail.com>

### Ⅱ. Does this pull request fix one issue?

fixes #2174 

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: charts/alluxio/templates/fuse/daemonset.yaml [hunk 1]
```diff
@@ -51,6 +51,10 @@ spec:
         {{- end }}
         {{- end }}
     spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
       {{- if .Values.fuse.criticalPod }}
       priorityClassName: system-node-critical
       {{- end }}
```
Comment (cheyang) [inline charts/alluxio/templates/fuse/daemonset.yaml:60]: Please also update CHANGELOG and chart version.

**Review by cheyang — APPROVED** [review]

File: pkg/utils/docker/image.go [hunk 2]
```diff
@@ -56,6 +58,16 @@ func GetImageTagFromEnv(envName string) (tag string) {
 	return
 }
 
+// get docker pull secrets from environment variables, if it's not existed, return ""
+func GetImagePullSecretsFromEnv(envName string) (pullSecretsStr string) {
+	if value, existed := os.LookupEnv(envName); existed {
+		if existed && len(value) > 0 {
```
Comment (TrafalgarZZZ) [inline pkg/utils/docker/image.go:64]: I think it's a repeated `existed=true` check here

File: pkg/ddc/alluxio/transform.go [hunk 3]
```diff
@@ -127,7 +129,10 @@ func (e *AlluxioEngine) transformCommonPart(runtime *datav1alpha1.AlluxioRuntime
 	imageTag := runtime.Spec.AlluxioVersion.ImageTag
 	imagePullPolicy := runtime.Spec.AlluxioVersion.ImagePullPolicy
 
-	value.Image, value.ImageTag, value.ImagePullPolicy = e.parseRuntimeImage(image, imageTag, imagePullPolicy)
+	// TODO: support imagePullSecrets by AlluxioRuntime
+	imagePullSecrets := []corev1.LocalObjectReference{}
+
+	value.Image, value.ImageTag, value.ImagePullPolicy, value.ImagePullSecrets = e.parseRuntimeImage(image, imageTag, imagePullPolicy, imagePullSecrets)
```
Comment (TrafalgarZZZ) [inline pkg/ddc/alluxio/transform.go:135]: Why should we pass an empty `imagePullSecrets` object into the function?

Comment (allenhaozi) [reply pkg/ddc/alluxio/transform.go:135]: > Why should we pass an empty `imagePullSecrets` object into the function?

Prepare for the subsequent support of pass image pull secret directly through Alluxioruntime

File: pkg/utils/docker/image.go (hunk 2 above)
Comment (allenhaozi) [reply pkg/utils/docker/image.go]: > I think it's a repeated `existed=true` check here

thank you, done

**Pull request merged**